src/contracts/core/StakeRootCompendium.sol

```diff
@@ -189,29 +216,34 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
         uint256 indexChargePerProof,
         Proof calldata _proof
     ) external {
-        require(calculationTimestamp % proofIntervalSeconds == 0, "StakeRootCompendium._postStakeRoot: timestamp must be a multiple of proofInterval");
+        require(
+            calculationTimestamp % cumulativeChargeParams.proofIntervalSeconds == 0,
+            TimestampNotMultipleOfProofInterval()
+        );
         // no length check here is ok because the initializer adds a default submission
         require(
-            stakeRootSubmissions[stakeRootSubmissions.length - 1].calculationTimestamp != calculationTimestamp, 
-            "StakeRootCompendium._postStakeRoot: timestamp already posted"
+            stakeRootSubmissions[stakeRootSubmissions.length - 1].calculationTimestamp != calculationTimestamp,
+            TimestampAlreadyPosted()
         );
         // credit the charge recipient
-        Snapshots.Snapshot memory _snapshot = totalChargeHistory._snapshots[indexChargePerProof];
-        require(_snapshot._key <= calculationTimestamp, "StakeRootCompendium._postStakeRoot: timestamp of indexChargePerProof is greater than the calculationTimestamp");
+        Snapshots.Snapshot memory _snapshot = chargePerProof._snapshots[indexChargePerProof];
```

Comment:
i much prefer `totalChargeHistory`, `chargePerProof` doesn't make it clear that it's a proof across the entire stakeTree

Reply:
I put this in to disambiguate totalCharge and cumulativeCharge which as synonyms with different meanings.

> doesn't make it clear that it's a proof across the entire stakeTree

but this contract does not use charge for multiple proof types . there is nothing to disambiguate. We discussed a fallback for OperatorSets to generate OperatorSet specific proofs, but we don't plan to introduce a billing layer there right?

If you have a strong preference we can change it back ofc

Reply:
i think this is as opposed to _charge per proof that each operatorSet pays_

agree the total/cumulative is gross. we can change it for now, noted in stake-root PR comment